src/api/createCpmProduct/src/v1/steps.py

```diff
@@ -0,0 +1,68 @@
+from http import HTTPStatus
+
+from aws_lambda_powertools.utilities.data_classes import APIGatewayProxyEvent
+from domain.core.cpm_product import CpmProduct, CpmProductIncomingParams
+from domain.core.cpm_system_id import ProductId
+from domain.core.product_team.v3 import ProductTeam
+from domain.repository.product_team_repository.v2 import ProductTeamRepository
+from domain.response.validation_errors import (
+    InboundValidationError,
+    mark_json_decode_errors_as_inbound,
+)
+from event.aws.client import dynamodb_client
+from event.step_chain import StepChain
+from pydantic import ValidationError
+
+
+@mark_json_decode_errors_as_inbound
+def parse_event_body(data, cache) -> dict:
+    event = APIGatewayProxyEvent(data[StepChain.INIT])
+    return event.json_body if event.body else {}
+
+
+def parse_incoming_cpm_product(data, cache) -> CpmProductIncomingParams:
+    json_body = data[parse_event_body]
+    try:
+        incoming_product = CpmProductIncomingParams(**json_body)
+        return incoming_product
+    except ValidationError as exc:
+        raise InboundValidationError(errors=exc.raw_errors, model=exc.model)
+
+
+def read_product_team(data, cache) -> ProductTeam:
+    incoming_product = data[parse_incoming_cpm_product]
+    product_team_repo = ProductTeamRepository(
+        table_name=cache["DYNAMODB_TABLE"], dynamodb_client=dynamodb_client()
+    )
+    return product_team_repo.read(id=incoming_product.product_team_id)
+
+
+def generate_cpm_product_id(data, cache) -> str:
+    generated_product_id = ProductId.create()
+    return generated_product_id.latest_id
```

Comment:
Just a point for discussion in that this looks a little weird as it sounds like we are tracking it - I know its just cause of the base class but wanted to float it anyway

Reply:
return generated_product_id.id